### ISSUE TITLE
Signature mismatch

### ISSUE THREAD

**Issue by wilsonsilva** [leostera/atacama#1]
Hello. Apologies for the dumb question. I'm trying to run the atacama example in the `bench` folder by running `dune build` and I get this error:

```sh
> $ dune build                                                                                                                                                                                                
Entering directory '/Users/wilson/projects/github/leostera/atacama'
File "atacama/transport.ml", lines 43-52, characters 20-3:
43 | ....................struct
44 |   let listen = Net.Socket.listen
45 |   let connect = Net.Socket.connect
46 |   let accept = Net.Socket.accept
47 |   let close = Net.Socket.close
48 |   let controlling_process = Net.Socket.controlling_process
49 |   let receive = Net.Socket.receive
50 |   let send = Net.Socket.send
51 |   let handshake _socket = Ok ()
52 | end
Error: Signature mismatch:
       ...
       Values do not match:
         val receive :
           ?timeout:Net.Socket.timeout ->
           len:int ->
           Net.Socket.stream_socket ->
           (Bigstringaf.t,
            [> `Closed | `Timeout | `Unix_error of Unix.error ])
           result
       is not included in
         val receive :
           ?timeout:Net.Socket.timeout ->
           buf:Bigstringaf.t ->
           Net.Socket.stream_socket ->
           (Bigstringaf.t,
            [> `Closed | `Timeout | `Unix_error of Unix.error ])
           result
       The type
         ?timeout:Net.Socket.timeout ->
         len:int ->
         Net.Socket.stream_socket ->
         (Bigstringaf.t, [> `Closed | `Timeout | `Unix_error of Unix.error ])
         result
       is not compatible with the type
         ?timeout:Net.Socket.timeout ->
         buf:Bigstringaf.t ->
         Net.Socket.stream_socket ->
         (Bigstringaf.t, [> `Closed | `Timeout | `Unix_error of Unix.error ])
         result
       Type
         len:int ->
         Net.Socket.stream_socket ->
         (Bigstringaf.t, [> `Closed | `Timeout | `Unix_error of Unix.error ])
         result
       is not compatible with type
         buf:Bigstringaf.t ->
         Net.Socket.stream_socket ->
         (Bigstringaf.t, [> `Closed | `Timeout | `Unix_error of Unix.error ])
         result
       File "atacama/transport.ml", lines 28-32, characters 2-62:
         Expected declaration
       File "atacama/transport.ml", line 49, characters 6-13:
         Actual declaration
```

I have never used OCaml before. Perhaps there's a better way to do this? In Go there's `go run`, in Rust there's `cargo run` and in Elixir there's just `elixir`.

**Comment by leostera** [comment]
Hi @wilsonsilva! 👋🏼 thanks for opening the PR.

At the moment the main branch of Atacama relies on the main branch of Riot, so you have to pin [this](https://github.com/leostera/riot) repository if you're using opam+dune.

Riot 0.0.6 will be submitted soon and then Atacama will be buildable on its own 🚀

**Comment by leostera** [comment]
This should be fixed in the upcoming 0.0.4 release. If you're working from sources you'll have to keep Riot pinned. Closing for now :) please reopen if the issue appears again!
